dev/benchmarks/macrobenchmarks/test/util.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/scheduler.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:macrobenchmarks/common.dart';
+import 'package:e2e/e2e.dart';
+import 'package:macrobenchmarks/main.dart' as app;
+
+/// The maximum amount of time considered safe to spend for a frame's build
+/// phase. Anything past that is in the danger of missing the frame as 60FPS.
+Duration kBuildBudget = const Duration(milliseconds: 16);
+
+const String _kDebugWarning = '''
+┏╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍┓
+┇ ⚠    THIS BENCHMARK IS BEING RUN IN DEBUG MODE     ⚠  ┇
+┡╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍┦
+│                                                       │
+│  Numbers obtained from a benchmark while asserts are  │
+│  enabled will not accurately reflect the performance  │
+│  that will be experienced by end users using release  ╎
+│  builds. Benchmarks should be run using this command  ┆
+│  line:  flutter drive --profile test_perf.dart        ┊
+│                                                       ┊
+└─────────────────────────────────────────────────╌┄┈  🐢
+''';
+
+void macroPerfTest(
+  String testName,
+  String routeName, {
+  Duration pageDelay,
+  Duration duration = const Duration(seconds: 3),
+  Duration timeout = const Duration(seconds: 30),
+  Future<void> driverOps(WidgetController controller),
```

Comment:
I interpret `driver` here as the operation to *drive* the test, rather than from `flutter_driver`. What about call them `driverCallback` and `setupCallback`. And maybe i should also `typedef` the callback.

Reply:
Done